src/lib/common/topology_sampler.cpp

```diff
@@ -3,7 +3,9 @@
 #include "semphr.h"
 #include "task.h"
 #include "task_priorities.h"
+extern "C" {
 #include "timer_callback_handler.h"
+}
```

Comment:
This feels off-putting to put in the top of so many files, I get why we were doing it for BCMP stuff because all of bristlemouth stack has not been c-ified, but it feels awkward here, should we place this directive in the `timer_callback_handler.h` file? @victorsowa12  @towynlin ??

Reply:
Either one is fine with me. I'll move it over to the header file.